### ISSUE TITLE
Progress

### ISSUE THREAD

**Issue by jp-praveen** [jp-praveen/Rendezvous#17]
Dear Sir,
       I was sick yesterday, so I couldn't do much of the work. I will start doing the things which we discussed and send you the graphs as soon as possible.

**Comment by joelgeorgem** [comment]
Okay. Take care.

**Comment by jp-praveen** [comment]
Dear Sir,
    I have incorporated the SGP4 code on our code and we no longer need to input the position and velocity data. I have also plotted the graphs, coasting time vs transfer time, which are attached below,  
![chaer-debri1](https://user-images.githubusercontent.com/55296086/66651702-52f7fa00-ec51-11e9-87cb-738121b23adf.jpg)
![chaer-debri2](https://user-images.githubusercontent.com/55296086/66651703-52f7fa00-ec51-11e9-832a-c9f6f319999d.jpg)
![chaer-debri3](https://user-images.githubusercontent.com/55296086/66651704-53909080-ec51-11e9-8dd8-00a7fdd38bc5.jpg)
![chaer-debri4](https://user-images.githubusercontent.com/55296086/66651705-53909080-ec51-11e9-90da-1d6897f3c233.jpg)
![chaer-debri5](https://user-images.githubusercontent.com/55296086/66651707-53909080-ec51-11e9-99a1-0cd1f6091231.jpg)

**Comment by joelgeorgem** [comment]
Maybe, we should reduce &Delta;V.

**Comment by jp-praveen** [comment]
Sir, When I reduced ΔV to 2Km/s, transfer trajectories are possible only to the 2nd and 5th debris. Transfer trajectories to the other debris are obtained only when I input ΔV>13Km/s

**Comment by joelgeorgem** [comment]
What are the typical &Delta;V values for debris to debris transfer given in those papers?

**Comment by jp-praveen** [comment]
They have used 3 cases [dv=0.5km/s, dv=0.75km/s, dv=1km/s]. I think while generating the transfer trajectories they have also encountered this problem. Under Generating Database for Rendezvous trajectories, it is given that, 
"Postprocessing to remove expensive trajectories was conducted to reduce the size of the database based on two pruning conditions. The first pruning condition is the ΔV value of the rendezvous. Figures 6 and 7 show the minimum ΔV for every departure/arrival debris pair over the entire mission duration for short-term and long-term missions, respectively (the darker color represents the smaller ΔV). **Only a fraction of each debris pair is transferable with a reasonable ΔV budget, and the trajectory whose rendezvous ΔV is higher than a predetermined value(set up as1 km ∕s in this study)can be discarded.** Second, a rendezvous trajectory dominated by another one in the same solution set is eliminated. Let τ1 and τ2 be the local optimal solutions of a single-target rendezvous problem associated with two debris. If τ1 has earlier departure time td, later arrival time ta, and larger associated ΔV compared with τ2 simultaneously, τ1 is dominated by τ2 and can be discarded (not included in the final database)".

**Comment by joelgeorgem** [comment]
What are the typical transfer times from one debris to another?

**Comment by jp-praveen** [comment]
Sir, It varies depending on the given dv max. For ΔV=1km/s, transfer time to 2nd debris varies from 150s to 650s depending on the initial position of the chaser and the debris. For high ΔV like ΔV=10km/s, the transfer time typically takes 50s mostly.

**Comment by joelgeorgem** [comment]
Do the papers use &Delta;V of the order of 10km/s? I am asking what are typical transfer times that the papers that we are referring to use.

**Comment by jp-praveen** [comment]
Sir, They typically have transfer time in the order of 0.5 days and their dv come in the order of 0.1km/s.

**Comment by joelgeorgem** [comment]
Let us also go for higher transfer times and lower &Delta;V if that helps. &Delta;V of 10km/s and all are totally infeasible. I don't know how low we will be able to go; plane change maneuvers are costly.

**Comment by jp-praveen** [comment]
Yes sir, I will run for higher transfer time and take the debris which they considered and check how low we can get.

**Comment by jp-praveen** [comment]
Sir, I ran the code for a few other debris and some of them had high  ΔV  and some had  ΔV in the range of 1 km/sec. Also, I tried increasing the transfer time, but the  ΔV didn't get reduced much. It stayed on the same range.

**Comment by jp-praveen** [comment]
Dear Sir, 
          Can we extend the heuristic method to the A* method? maybe we can take only those debris whose ΔV's are feasible.

**Comment by joelgeorgem** [comment]
Let us meet tomorrow at 11 am? 

Why is it that &Delta;V is small for the papers? Because they are not using Lambert manoeuvre? 

Get few plots of how &Delta;V varies with transfer time for transfer between few of the debris pairs. 

Also, we need to pick a paper with which we can compare the performance of our algorithm.

**Comment by jp-praveen** [comment]
Yes Sir, I will get the plots and meet you tomorrow. The paper is also using the Lambert maneuver.

**Comment by joelgeorgem** [comment]
Please attach the paper.

**Comment by jp-praveen** [comment]
[Multitarget Rendezvous for Active Debris Removal Using Multiple Spacecraft.pdf](https://github.com/jp-praveen/Rendezvous/files/3727548/Multitarget.Rendezvous.for.Active.Debris.Removal.Using.Multiple.Spacecraft.pdf)
[Optimization of time open constrained Lambert rendezvous using interval analysis.pdf](https://github.com/jp-praveen/Rendezvous/files/3727550/Optimization.of.time.open.constrained.Lambert.rendezvous.using.interval.analysis.pdf)

These are the papers sir.

**Comment by joelgeorgem** [comment]
Tomorrow also I need to go to the hospital.

I will let you know when we can meet again.

**Comment by jp-praveen** [comment]
Yes sir. Take care.
Meanwhile, I will search for other journals to compare our ΔV and find out why we are getting high ΔV values.

**Comment by joelgeorgem** [comment]
**Only a fraction of each debris pair is transferable with a reasonable ΔV budget**: can we find those pairs?

**Comment by jp-praveen** [comment]
Yes sir. We can run the code for many debris, like 50 or so, and then find those pairs.

**Comment by joelgeorgem** [comment]
Okay. Search over 50 or 100 (as in the paper) and find those "handful" of pairs.

**Comment by jp-praveen** [comment]
Yes sir. I will do that.

**Comment by joelgeorgem** [comment]
Any progress?

**Comment by jp-praveen** [comment]
Yes Sir. I actually encountered a problem while taking the debris data. I took the current data of the debris which they used. Actually, we need to take the data as of December 1, 2017, that was the time they took the data. I am redoing that part and nearly completed it. I will send them soon.

**Comment by jp-praveen** [comment]
[matrix.xlsx](https://github.com/jp-praveen/Rendezvous/files/3748313/matrix.xlsx)
Sir, 
The matrix contains all transfers with ΔV <1.2 km/sec. I uploaded the excel version but will make a Matlab file also.

**Comment by jp-praveen** [comment]
Dear Sir,
 I made the .mat file and updated the same on Git.

**Comment by jp-praveen** [comment]
Dear Sir, 
       Will the next step in the project be making an A* Algorithm?

**Comment by joelgeorgem** [comment]
Yes, go ahead.

**Comment by joelgeorgem** [comment]
Which algorithm did you use to compute &Delta;Vs? Curtis or Prussing?

**Comment by jp-praveen** [comment]
I used the Prussings algorithm sir.

**Comment by joelgeorgem** [comment]
Provide me with the sketch of the algorithm that you are planning to implement.

**Comment by jp-praveen** [comment]
Sir, I modified the a* algorithm and made an algorithm that will fill a table with 1st-row corresponding to the cost or dv and 2nd row corresponding to the parent debris. There are 2 vectors open and close. The open will have all debris that is not visited. Once a debris is visited and all the possible paths from it are calculated it is removed from open and put in close. A rough algorithm is as follows,(I didn't check whether the algorithm works, but I started to implement it)

[check.txt](https://github.com/jp-praveen/Rendezvous/files/3762230/check.txt)

We can get the cost and sequence from the table
